feat(build): ignore out-of-source build folder (#33191)
neovim/neovim: CMakeLists.txt

include(PreventInTreeBuilds)
include(Util)

if(NOT PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)

>   # Git auto-ignore out-of-source build directory

Reviewer: ```suggestion
  # Auto-create a .gitignore in the specified "build" directory.
```

Author: Fixed
